src/test/java/org/smartregister/converters/ClientConverterTest.java

```diff
@@ -21,19 +22,43 @@
 	@Test
 	public void testConvertToPatientResource() {
 		Client client = gson.fromJson(CLIENT_JSON, Client.class);
-				client.setBirthdate(new DateTime(0l));
-				client.setDeathdate(new DateTime(0l));
-				client.setFirstName("John");
-				client.setMiddleName("Lewis");
-				client.setLastName("Johny");
-				client.setGender("male");
+		client.setBirthdate(new DateTime(0l));
+		client.setDeathdate(new DateTime(0l));
+		client.setFirstName("John");
+		client.setMiddleName("Lewis");
+		client.setLastName("Johny");
+		client.setGender("male");
 		Patient patient = ClientConverter.convertClientToPatientResource(client);
 		assertNotNull(patient);
-		assertEquals(patient.getGender().getValueAsEnumConstant().value(),client.getGender());
-		assertEquals(patient.getName().get(0).getFamily().getValue(),client.getLastName());
-		assertEquals(patient.getName().get(0).getText().getValue(),client.fullName());
-		assertEquals(patient.getId(),client.getBaseEntityId());
-		//TODO : Add assertion on remaining properties
+		assertEquals(patient.getGender().getValueAsEnumConstant().value(), client.getGender());
+		assertEquals(patient.getName().get(0).getFamily().getValue(), client.getLastName());
+		assertEquals(patient.getName().get(0).getText().getValue(), client.fullName());
+		assertEquals(patient.getId(), client.getBaseEntityId());
+		int identifiersSize = client.getRelationships().entrySet().size() + client.getAttributes().entrySet().size() + client
+				.getIdentifiers().entrySet().size();
+		assertEquals(patient.getIdentifier().size(), identifiersSize);
+		assertEquals(patient.getBirthDate().getValue().toString(), client.getBirthdate().toString("YYYY-MM-DD"));
+		assertEquals(patient.getDeceased().as(com.ibm.fhir.model.type.DateTime.class).getValue().toString(),
+				client.getDeathdate().toString("yyyy-MM-dd'T'HH:mm'Z'"));
+		for (Identifier identifier : patient.getIdentifier()) {
+			if (identifier.getSystem() != null && identifier.getSystem().getValue()
```

Comment:
Use fhirpath expressions for lookup as if the identifier does not exist then test will fail. With this approach of looping  tests will fail if identifiers don't exist

Reply:
check this https://github.com/OpenSRP/opensrp-plan-evaluator/pull/28/files#r443378880

Reply:
Updated.